internal/eval/eval.go

```diff
@@ -2832,3 +2832,40 @@ func evalSELECT(args []string, store *dstore.Store) []byte {
 
 	return clientio.RespOK
 }
+
+func evalSETEX(args []string, store *dstore.Store) []byte {
+	if len(args) != 3 {
+		return diceerrors.NewErrArity("SETEX")
+	}
+
+	var key, value string
+	var exDurationMs int64
+
+	key, value = args[0], args[2]
+	oType, oEnc := deduceTypeEncoding(value)
+
+	exDuration, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		return diceerrors.NewErrWithMessage(diceerrors.IntOrOutOfRangeErr)
+	}
+	if exDuration <= 0 || exDuration >= maxExDuration {
+		return diceerrors.NewErrExpireTime("SETEX")
+	}
+	exDurationMs = exDuration * 1000
+
+	// Cast the value properly based on the encoding type
+	var storedValue interface{}
+	switch oEnc {
+	case object.ObjEncodingInt:
+		storedValue, _ = strconv.ParseInt(value, 10, 64)
+	case object.ObjEncodingEmbStr, object.ObjEncodingRaw:
+		storedValue = value
+	default:
+		return clientio.Encode(fmt.Errorf("ERR unsupported encoding: %d", oEnc), false)
+	}
+	// putting the k and value in a Hash Table
+	store.Put(key, store.NewObj(storedValue, exDurationMs, oType, oEnc))
+
+	return clientio.RespOK
+
+}
```

Comment:
Pardon me if this sounds silly, but what do you feel about the following code? It ensures SETEX follows all the conventions of SET in dice, and any addition of encodings in the future can be addressed. The Redis documentation states `SETEX key seconds value` can be transformed to `SET key value ex seconds`
```suggestion
	if len(args) != 3 {
		return diceerrors.NewErrArity("SETEX")
	}

	var key, exDuration, value string
	key, exDuration, value = args[0], args[1], args[2]
	parseArgs := []string{key, value,ex,exDuration}
	return evalSET(parseArgs, store)

}
```